boa/src/builtins/object/gcobject.rs

```diff
@@ -151,19 +160,24 @@ impl GcObject {
 
                         ctx.realm.environment.push(local_env);
 
-                        // Call body should be set before reaching here
-                        let result = body.run(ctx);
-
-                        // local_env gets dropped here, its no longer needed
-                        ctx.realm.environment.pop();
-                        result
+                        FunctionBody::Ordinary(RcStatementList::from(body.clone()))
```

Comment:
It's better to expose the `RcStatementList` rather than copy the whole function body ast every time we call.